etcdmain/config.go

```diff
@@ -317,6 +356,141 @@ func (cfg *config) Parse(arguments []string) error {
 		}
 	}
 
+	return nil
+}
+
+func (cfg *config) configFromFile() error {
+	// Read in the contents of the config file.
+	b, err := ioutil.ReadFile(cfg.configFile)
+	if err != nil {
+		return err
+	}
+
+	// Unmarshal the YAML.
+	err = yaml.Unmarshal(b, cfg)
+	if err != nil {
+		return err
+	}
+
+	// Make sure the defaults are set for the missing fields.
+	cfg.setDefaultsIfNeeded()
+
+	// Validate whether conflicting parameters are specified.
+	nSet := 0
```

Comment:
> Can you please let me know what is your thinking on moving the common validations to Parse function? I mean how do we handle them for both cases (cmd line and config file) as the flags.IsSet won't work for the configFile case.

For the validation, I think we can make a function and let it take isSet as args. For conf and flag, we probably have different way to test if it is set. I am not sure if it will be better than what we have today. I basically want to put the validation logic at the same place. So if we change the validation part, we do not need to repeat ourselves.